app/src/main/java/io/appium/settings/RecorderConstant.java

```diff
@@ -0,0 +1,38 @@
+/*
+  Copyright 2012-present Appium Committers
+  <p>
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  <p>
+  http://www.apache.org/licenses/LICENSE-2.0
+  <p>
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package io.appium.settings;
+
+public class RecorderConstant {
+    public static final int REQUEST_CODE_SCREEN_CAPTURE = 123;
+    private static final String PACKAGE_NAME = BuildConfig.APPLICATION_ID;
+    public static final String ACTION_RECORDING_START = PACKAGE_NAME + ".recording.ACTION_START";
+    public static final String ACTION_RECORDING_STOP = PACKAGE_NAME + ".recording.ACTION_STOP";
+    public static final String ACTION_RECORDING_RESULT_CODE = "result_code";
+    public static final String ACTION_RECORDING_FILENAME = "recording_filename";
+    public static final String ACTION_RECORDING_ROTATION = "recording_rotation";
```

Comment:
It still makes sense to make values like:
- bitrate
- scale
- FPS 
configurable. They affect the size of the resulting video and we don't always want it to be too huge.

I also usually add an automatic timer to such recorders to prevent the recording to continue forever and eventually exceed the free sdcard space in case of a client failure.

Reply:
maybe we also need a separate flag on whether to disable the audio track or video track

Reply:
> It still makes sense to make values like:
> 
> * bitrate
> * scale
> * FPS
>   configurable. They affect the size of the resulting video and we don't always want it to be too huge.
> 
> I also usually add an automatic timer to such recorders to prevent the recording to continue forever and eventually exceed the free sdcard space in case of a client failure.

well we are not and dont have scaling right now (we are using device's default width/height) scaling will bring complexity since we need to somehow transform images on-the-buffer (it will also slow down recording and may cause jitters probably) this can be future-direction, we can add as a feature on top of recording in the future maybe 👍 
_for bitrate_: since we also calculate bitrate dynamically and use width/height to calculate it, there is no large room for fine-tuning bitrate (only bitrate multiplier which can be dynamic) or else we will allow full customization without boundaries? 
_for fps_: yeah i will take a look at it to make it dynamic let's see how it goes 👍

Reply:
> I also usually add an automatic timer to such recorders to prevent the recording to continue forever and eventually exceed the free sdcard space in case of a client failure.

+1 good point, instead of timer, what about storage size limitation? when i looked into limitations of storage on Android, i have found out that some says 3GB is the limit https://stackoverflow.com/a/67017220 and another solution is to check avail size dynamically https://stackoverflow.com/a/20329027

Reply:
Regarding the features I'm more than ok to do in in different PRs as this one is anyway quite big and it is not a very easy task to go over the whole source code again and again for each review.

Reply:
> maybe we also need a separate flag on whether to disable the audio track or video track

+1000 yeah i was thinking the same, since we are already writing audio and video buffers separately, we can control it via flags to produce 

- audio-only -> m4a or aac
- video-only -> mp4
- audio-video mixed -> mp4, default

Reply:
> Altogether @sirmordred I would like to thank you for the great work and all the efforts invested into this PR. Let me know if/when you decide to look for new job opportunities ;-)

thank you so much for your kind answers and great direction and review 💯  it was pleasure for me to work with you 👍  and really thanks for your job opportunities, i really want to work with you guys and contribute to appium and its infrastructure (lets keep connection through mail, i'll mail it asap 👍 )

Reply:
done, added maximum allowed duration which is default 15 minute but it can be also configurable by user 👍 
 https://github.com/sirmordred/io.appium.settings/commit/afa8fbcef223013cc6c46a9442721085e4e82d18